File: Android App - Shredsheets 2/app/src/main/java/revert/shredsheets/models/scales/Pentatonic.java
package revert.shredsheets.models.scales;

public class Pentatonic extends Scale {

    @Override
    public int[] getScaleIntervals() {
        return new int[]{ 3, 2, 2, 3, 2 };
    }

    @Override
    public String getName() {
        return "Pentatonic";
    }

    @Override
    public String[] getModeNames() {
        return new String[]{ "Pentatonic Minor","Pentatonic Major", "Suspended Pentatonic", "Man Gong", "Ritusen" };
    }

    @Override
    protected boolean[][] getModalHighlighting() {
        return new boolean[][]{
                {true, true, false, true, true},
                {true, false, true, true, false},
                {true, false, false, true, true},
                {true, true, false, false, true},
                {true, false, false, true, false},
         };
    }


    @Override
    public String[] getChords() {
        return new String[] { "m7", "M", "5 ♭7", "♭3 ♭7", "5" };
    }

    @Override
    public int[][] getDegrees() {
        return new int[][] {
                { 1, 3, 4, 5, 7 },
                { 1, 2, 3, 5, 6 },
                { 1, 2, 4, 5, 7 },
                { 1, 3, 4, 6, 7 },
                { 1, 2, 4, 5, 6 }
        };
    }
}

File: Shredsheets 3/core/src/com/bm/shredsheets/models/scales/Pentatonic.kt
package com.bm.shredsheets.models.scales

class Pentatonic : Scale() {
    override val scaleIntervals: IntArray
        get() = intArrayOf(3, 2, 2, 3, 2)

    override val name: String
        get() = "Pentatonic"

    override fun getModeNames(): Array<String> {
        return arrayOf("Pentatonic Minor", "Pentatonic Major", "Suspended Pentatonic", "Man Gong", "Ritusen")
    }

    override val getEnum: Scales
        get() = Scales.Pentatonic

    override val chords: Array<String>
        get() = arrayOf("m7", "M", "5 ♭7", "♭3 ♭7", "5")

    override val degrees: Array<IntArray>
        get() = arrayOf(
            intArrayOf(1, 3, 4, 5, 7),
            intArrayOf(1, 2, 3, 5, 6),
            intArrayOf(1, 2, 4, 5, 7),
            intArrayOf(1, 3, 4, 6, 7),
            intArrayOf(1, 2, 4, 5, 6))
}